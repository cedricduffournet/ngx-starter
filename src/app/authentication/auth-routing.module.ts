import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent
} from '@app/authentication/containers';
import { authConfig } from '@app/authentication/auth.config';

const AUTHENTICATION_ROUTES: Routes = [
  { path: 'login', component: LoginViewComponent },
  { path: 'resetting', component: SendResetPasswordViewComponent },
  {
    path: `${authConfig.path.passwordResetConfirm}/:token`,
    component: ResetPasswordViewComponent
  },
  { path: 'register', component: RegisterViewComponent },
  {
    path: `${authConfig.path.registerConfirm}/:token`,
    component: RegisterConfirmViewComponent
  }
];

@NgModule({
  imports: [RouterModule.forChild(AUTHENTICATION_ROUTES)],
  exports: [RouterModule]
})
export class AuthRoutingModule {}
