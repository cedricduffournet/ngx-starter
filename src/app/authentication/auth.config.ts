export const authConfig = {
  path: {
    registerConfirm: 'register/confirm',
    passwordResetConfirm: 'password/reset'
  },
  storageKey: 'authentication'
};
