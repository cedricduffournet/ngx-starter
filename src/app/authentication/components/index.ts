export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
