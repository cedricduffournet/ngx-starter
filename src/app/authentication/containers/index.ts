export * from './login-view/login-view.component';
export * from './register-confirm-view/register-confirm-view.component';
export * from './register-view/register-view.component';
export * from './reset-password-view/reset-password-view.component';
export * from './send-reset-password-view/send-reset-password-view.component';
export * from './logout-modal/logout-modal.component';
