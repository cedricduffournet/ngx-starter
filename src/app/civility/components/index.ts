export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
