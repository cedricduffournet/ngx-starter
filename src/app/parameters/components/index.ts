export * from './referenced-table-menu/referenced-table-menu.component';
