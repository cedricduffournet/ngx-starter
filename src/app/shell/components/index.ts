export * from './navbar/navbar.component';
export * from './dropdown-navbar/dropdown-navbar.component';
export * from './menu/menu.component';
