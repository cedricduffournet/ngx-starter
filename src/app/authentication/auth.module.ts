import { NgModule } from '@angular/core';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityModule } from '@app/civility/civility.module';
import { LangSwitcherModule } from '@app/shared/lang-switcher/lang-switcher.module';
import { LogoModule } from '@app/shared/logo/logo.module';
import { AuthRoutingModule } from '@app/authentication/auth-routing.module';
import { AuthEffects } from '@app/authentication/state/effects';
import { reducers } from '@app/authentication/state/reducers';
import { AuthFacade } from '@app/authentication/state/auth.facade';

import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent,
  LogoutModalComponent
} from '@app/authentication/containers';

import {
  AuthLayoutComponent,
  LoginFormComponent,
  ResetPasswordFormComponent,
  SendResetPasswordFormComponent,
  RegisterFormComponent
} from '@app/authentication/components';

@NgModule({
  imports: [
    AuthRoutingModule,
    SharedModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects]),
    LangSwitcherModule,
    LogoModule,
    CivilityModule
  ],
  declarations: [
    AuthLayoutComponent,
    LoginViewComponent,
    SendResetPasswordViewComponent,
    ResetPasswordViewComponent,
    RegisterViewComponent,
    RegisterConfirmViewComponent,
    LoginFormComponent,
    ResetPasswordFormComponent,
    SendResetPasswordFormComponent,
    LogoutModalComponent,
    RegisterFormComponent
  ],
  entryComponents: [LogoutModalComponent],
  providers: [AuthFacade]
})
export class AuthModule {}
