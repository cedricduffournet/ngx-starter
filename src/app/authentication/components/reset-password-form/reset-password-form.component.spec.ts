import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { ResetPasswordFormComponent } from '@app/authentication/components';

describe('ResetPasswordForm', () => {
  let fixture: ComponentFixture<ResetPasswordFormComponent>;
  let component: ResetPasswordFormComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule, TranslateModule.forRoot(), FormModule],
      declarations: [ResetPasswordFormComponent]
    });
    fixture = TestBed.createComponent(ResetPasswordFormComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should not emit event if missing password', () => {
    const plainPassword = {
      first: '',
      second: 'test'
    };

    component.resetPasswordForm.setValue({ plainPassword });

    spyOn(component.resetPassword, 'emit');
    component.onSubmit();

    expect(component.resetPassword.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if passwords are different', () => {
    const plainPassword = {
      first: '12345678',
      second: '87654321'
    };

    component.resetPasswordForm.setValue({ plainPassword });
    fixture.detectChanges();
    spyOn(component.resetPassword, 'emit');
    component.onSubmit();
    expect(component.resetPassword.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if passwords not conformed', () => {
    const plainPassword = {
      first: '1234',
      second: '1234'
    };

    component.resetPasswordForm.setValue({ plainPassword });

    spyOn(component.resetPassword, 'emit');
    component.onSubmit();
    expect(component.resetPassword.emit).not.toHaveBeenCalled();
  });

  it('should emit an event if the form is valid when submitted', () => {
    const plainPassword = {
      first: 'Test1234',
      second: 'Test1234'
    };

    component.resetPasswordForm.setValue({ plainPassword });

    spyOn(component.resetPassword, 'emit');
    component.onSubmit();

    expect(component.resetPassword.emit).toHaveBeenCalledWith({
      plainPassword
    });
  });
});
