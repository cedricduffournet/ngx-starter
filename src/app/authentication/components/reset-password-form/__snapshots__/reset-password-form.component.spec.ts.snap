// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm should be created 1`] = `
<app-reset-password-form
  formBuilder={[Function FormBuilder]}
  processing="false"
  resetPassword={[Function EventEmitter]}
  resetPasswordForm={[Function FormGroup]}
  submitted="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <div
      class="ng-untouched ng-pristine ng-invalid"
      formgroupname="plainPassword"
      ng-reflect-name="plainPassword"
    >
      <app-text-input
        class="ng-untouched ng-pristine ng-invalid"
        formcontrolname="first"
        ng-reflect-help-text="HELP_PASSWORD_FIELD"
        ng-reflect-label="PASSWORD"
        ng-reflect-name="first"
        ng-reflect-submitted="false"
        ng-reflect-type="password"
        type="password"
      >
        <div
          class="form-group "
        >
          <label
            class="control-label"
          >
            PASSWORD
          </label>
          <input
            class="form-control"
            placeholder=""
            type="password"
          />
          
          
          
          
          
          
          <small
            class="form-text text-muted"
          >
             HELP_PASSWORD_FIELD 
          </small>
        </div>
      </app-text-input>
      <app-text-input
        class="ng-untouched ng-pristine ng-invalid"
        formcontrolname="second"
        ng-reflect-error-text="FIELD_PASSWORD_MUST_BE_EQUAL"
        ng-reflect-label="PASSWORD_CONFIRMATION"
        ng-reflect-name="second"
        ng-reflect-type="password"
        type="password"
      >
        <div
          class="form-group "
        >
          <label
            class="control-label"
          >
            PASSWORD_CONFIRMATION
          </label>
          <input
            class="form-control"
            placeholder=""
            type="password"
          />
          
          
          
          
          
          
        </div>
      </app-text-input>
    </div>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-lg btn-block"
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         UPDATE_MY_PASSWORD 
      </button>
    </div>
  </form>
</app-reset-password-form>
`;
